concent_api/core/message_handlers.py

```diff
@@ -954,6 +958,58 @@ def store_message(
     return stored_message
 
 
+def handle_send_subtask_results_verify(
+    request,
+    subtask_results_verify: message.concents.SubtaskResultsVerify
+):
+    report_computed_task = subtask_results_verify.subtask_results_rejected.report_computed_task
+    compute_task_def = report_computed_task.task_to_compute.compute_task_def
+    validate_id_value(
+        compute_task_def['subtask_id'],
+        'subtask_id'
+    )
+    client_public_key = decode_client_public_key(request)
+    other_party_public_key = decode_other_party_public_key(request)
+
+    if Subtask.objects.filter(
+        Q(subtask_id=compute_task_def['subtask_id']),
+        Q(state=Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER.name) |  # pylint: disable=no-member
+        Q(state=Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name),  # pylint: disable=no-member
+    ).exists():
```

Comment:
I think this condition only checks if Subtask is currently in verification. In issue there is 

> Additional verification was done for this subtask before (ServiceRefused)

So I think there should be different check if verification was already done (please ask @cameel I don't really know how to check this). Or update issue description if it is wrong.

Reply:
I have just added a section that describes how to deal with duplicate requests: https://github.com/golemfactory/concent/issues/201#issue-308021944

We talked about it today but I realized that there are more cases so I have decided to include it in the issue.